plugins/inventory/d42.py

```diff
@@ -115,3 +114,537 @@ def parse(self, inventory, loader, path, cache=False):
+            left join (select device_fk, string_agg(view_devicenonauthoritativealias_v1.alias_name, ', ' order by view_devicenonauthoritativealias_v1.alias_name) as alias_names from view_devicenonauthoritativealias_v1 group by view_devicenonauthoritativealias_v1.device_fk) nonauthoritativealias
+                on view_device_v1.device_pk=nonauthoritativealias.device_fk
+            left join (select device_fk, string_agg(view_devicealias_v1.alias_name, ', ' order by view_devicealias_v1.alias_name) as alias_names from view_devicealias_v1 group by view_devicealias_v1.device_fk) alias
+                on view_device_v1.device_pk=alias.device_fk
+            left join (select device_pk, name from view_device_v1) vhost
+                on view_device_v1.virtual_host_device_fk=vhost.device_pk
+            left join (select objectcategory_pk, name from view_objectcategory_v1) obj_category
+                on view_device_v1.objectcategory_fk = obj_category.objectcategory_pk
+            left join (select vendor_pk, name from view_vendor_v1) vendor
+                on view_hardware_v1.vendor_fk = vendor.vendor_pk
+            left join (select customer_pk, name from view_customer_v1) customer
+                on view_device_v1.customer_fk = customer.customer_pk
+            LEFT OUTER JOIN (
+               SELECT p.device_fk,
+                 SUM(p.pcount *
+                     CASE
+                       WHEN pm.hdsize_unit = 'GB'
+                           THEN pm.hdsize
+                       WHEN pm.hdsize_unit = 'TB'
+                           THEN pm.hdsize * 1024
+                       WHEN pm.hdsize_unit = 'PB'
+                           THEN pm.hdsize * 1024 * 1024
+                       WHEN pm.hdsize_unit = 'EB'
+                           THEN pm.hdsize * 1024 * 1024 * 1024
+                       WHEN pm.hdsize_unit = 'ZB'
+                           THEN pm.hdsize * 1024 * 1024 * 1024 * 1024
+                       WHEN pm.hdsize_unit = 'YB'
+                           THEN pm.hdsize * 1024 * 1024 * 1024 * 1024 * 1024
+                       ELSE
+                           NULL
+                     END) AS total_part_disk_size
+                 FROM view_part_v1 p
+                 INNER JOIN view_partmodel_v1 pm ON p.partmodel_fk = pm.partmodel_pk
+                 WHERE pm.type_id = 3 AND pm.hdsize IS NOT NULL AND p.pcount > 0 AND p.device_fk IS NOT NULL
+                 GROUP BY p.device_fk
+             ) AS p ON view_device_v1.device_pk = p.device_fk
+        """
+
+        return self.get_doql_csv(device_query)
+
+
+    def get_custom_fields(self):
+
+        custom_field_query = """
+                select
+                view_device_v1.device_pk,
+                view_device_v1.name,
+                custom_field.key as custom_field_key,
+                custom_field.value as custom_field_value,
+                custom_field.notes as custom_field_notes
+                from view_device_v1
```

Comment:
I don't think the join to view_device_v1 is needed since view_device_custom_fields has device_fk.

Reply:
> I made all the queries perform a join strictly for device name. the name is not used when creating the json, but I wanted the queries to be reusable and be useful for debugging. if for any reason a device is not being imported support can call the same query and find what device information is not coming in for what device